### PR TITLE
Sprint 4 - Goal: To get only two topics complete in KYFB

### PR DESCRIPTION
**Pull request by DanielTurner**
Covers: C4A-204, CFA-195, CFA-193, C4A-194, CFA-185, CFA-247, CFA-248, CFA-249, CFA-251, CFA-186, CFA-189, CFA-190, CFA-191

Jira ticket links:
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-204
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-195
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-193
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-194
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-185
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-247
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-248
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-249
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-251
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-186
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-189
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-190
https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/C4A-191

This includes the updates from oct-22 pull request. They are detailed as follows.
* Image access issue (the images now should able to be accessed by the Queensland Health user)
* Questions now have an asterisk on the end of the question as to not interfere with the flow of the questions. These also use the pseudo-element after as they don't form part of the content
* These questions now have a 20px space above to provide more distinction between questions
* Navigation buttons now are fixed and are enabled/disabled correctly based on the form's state
* Tab indexing has been fixed, pressing the next button now also focuses the element at the top of the page so that the next tab key is the first link or button on the form depending on the context.
* Styling on 'hints' on questions and summary page is fixed for KYFB.

Changes completed since October 22nd are detailed as follows:
* Capitalisation of Summary changed on the download summary button
* Capitalisation of Summary changed on the email button
* Hyperlink added for contact your local council
* Added a missing full stop at the end of local council sentence
* Added aria-label for resource links
* Added in content for PDF for Animals and Pests and Health and Hygiene 
* Fixed vertical spacing on the summary page
* Added cancel button the home page for each topic
* Styled output PDF for health and hygiene and animals and pests
* Added validation with show hide for kyfb health and hygiene section
* Added in the ability to skip terms and conditions on the back button as well as the forward button
* Fixed up overlapping content on the PDF output
* Increased the clickable area on questions with radio buttons
* Generated email for based on content
* Fixed Spelling of topic
* Email content changes
* Terms of Use - Change colour of I agree text
* PDF feedback changes

URL:
https://www.qld.gov.au/dev/food-pantry/uat/know-your-food-business

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (duttonw) [inline LICENSE:1]: why not CC BY 3.0 AU or 4.0  as all www.qld.gov.au have https://creativecommons.org/licenses/by/4.0/ links at footer.

File: src/components/formio-wrapper.js [hunk 2]
```diff
@@ -20,6 +20,9 @@ export class FormioWrapper {
     this.termsConfig = configuration.termsConfig;
     this.formName = configuration.formName;
     this.extraTriggers = configuration.extraTriggersOnActions;
+    this.submissionInfo = configuration.submissionInfo;
+    this.submissionEndpoint = `https://api.forms.platforms.qld.gov.au/project/${this.submissionInfo.projectID}/form/${this.submissionInfo.formID}/submission`;
```
Comment (duttonw) [inline src/components/formio-wrapper.js:24]: 👍

File: src/components/formio-wrapper.js [hunk 3]
```diff
@@ -64,26 +67,43 @@ export class FormioWrapper {
...
+        this._downloadPDF();
+      });
+      this.wizard.on('sendEmail', () => {
+        this.wizard.data.sendEmail = true;
+        this._sendEmail();
+      });
+      this.wizard.on('nextPage', ({ page }) => {
+        if (page === 3) {
+          this.wizard.data.sendEmail = true;
+          this._sendEmail({ admin: true });
```
Comment (duttonw) [inline src/components/formio-wrapper.js:101]: admin true? what does that do?

**Review by ThrawnCA — APPROVED** [review]

File: src/components/formio-wrapper.js [hunk 4]
```diff
@@ -341,6 +375,17 @@ export class FormioWrapper {
     if (!this.loaded) {
       this.notLoaded();
     }
+    if (
+      this._shouldPreviousPageBeSkipped(this.wizard.page, this.wizard.pages)
+    ) {
+      const proposedPage = this.wizard.page - 2;
```
Comment (duttonw) [inline src/components/formio-wrapper.js:381]: - 2, what is this magic number doing?

File: src/components/topics-list.js [hunk 5]
```diff
@@ -29,9 +29,9 @@ export class TopicsList {
         name: 'Process',
         topics: [
           {
-            title: 'Recieve food',
+            title: 'Receive food',
             form:
-              'https://api.forms.platforms.qld.gov.au/fesrqwsyzlbtegd/recievefood',
+              'https://api.forms.platforms.qld.gov.au/fesrqwsyzlbtegd/receivefood',
```
Comment (duttonw) [inline src/components/topics-list.js:34]: should this be setting
{this.submissionInfo.projectID}
${this.submissionInfo.formID}

File: src/index.js [hunk 6]
```diff
@@ -43,6 +45,11 @@ import { ReapplySelected } from './scripts/reapply-selected';
     extraTriggersOnActions: {
       cancel: 'cancelKYFBForm',
     },
+    submissionInfo: {
+      projectID: '5f44969319d1a97a819d80a7',
```
Comment (duttonw) [inline src/index.js:49]: should we be using the internal id's instead of the nice alias names for form.io?

**Review by duttonw — APPROVED** [review]
Will wait til tuesday so others can code review

**Pull request merged**